sklearn/ensemble/_hist_gradient_boosting/tests/test_splitting.py

```diff
@@ -959,3 +959,86 @@ def test_split_interaction_constraints():
 
     # make sure feature 0 and feature 3 are split on in the constraint setting
     assert set(allowed_features) == set(split_features)
+
+
+@pytest.mark.parametrize("forbidden_features", [set(), {1, 3}])
+def test_split_feature_fraction_per_split(forbidden_features):
+    """Check that feature_fraction_per_split is respected."""
```

Comment:
```suggestion
    """Check that feature_fraction_per_split is respected.
    
    Because we set `n_features = 4` and `feature_fraction_per_split = 0.25`,
    it means that each call to `splitter.find_node_split` will be allowed to select
    a split for a single completely random feature at each call. So if we iterate
    enough, we should cover all the allowed features.
    """
```